telluric/georaster.py

```diff
@@ -907,8 +907,6 @@ def save(self, filename, tags=None, **kwargs):
                         r.write_mask(mask)
                         self._add_overviews_and_tags(r, tags, kwargs)
 
-                self._filename = filename  # Not sure about this, but the api made sure the internal filename is changed
-                # so the same object could be continue to use
                 return GeoRaster2.open(filename)
```

Comment:
I know this was introduced in https://github.com/satellogic/telluric/pull/197 by @guydou, but isn't it a bit strange that `raster.save` returns the raster being saved?